src/daft-local-execution/src/sources/scan_task.rs

```diff
@@ -159,11 +160,55 @@ async fn stream_scan_task(
             let inference_options =
                 ParquetSchemaInferenceOptions::new(Some(*coerce_int96_timestamp_unit));
 
-            if source.get_iceberg_delete_files().is_some() {
-                return Err(common_error::DaftError::TypeError(
-                    "Streaming reads not supported for Iceberg delete files".to_string(),
-                ));
-            }
+            let delete_rows = if let Some(delete_files) = source.get_iceberg_delete_files() {
+                let columns = Some(vec!["file_path".to_string(), "pos".to_string()]);
+                let delete_files = delete_files.to_owned();
+                let runtime = get_runtime(multi_threaded_io)?;
+                let io_client = io_client.clone();
+                let io_stats = io_stats.clone();
+                let tables = runtime.block_on_io_pool(async move {
+                    read_parquet_bulk_async(
+                        delete_files,
+                        columns,
+                        None,
+                        None,
+                        None,
+                        None,
+                        io_client,
+                        io_stats,
+                        *NUM_CPUS,
+                        inference_options,
+                        None,
+                        None,
+                        None,
+                        None,
+                    )
+                    .await
+                })??;
+
+                let mut delete_rows = vec![];
+                for table_result in tables {
+                    let table = table_result?;
+                    // values in the file_path column are guaranteed by the iceberg spec to match the full URI of the corresponding data file
+                    // https://iceberg.apache.org/spec/#position-delete-files
+                    let file_paths = table.get_column("file_path")?.downcast::<Utf8Array>()?;
+                    let positions = table.get_column("pos")?.downcast::<Int64Array>()?;
+
+                    for i in 0..table.len() {
+                        let file = file_paths.get(i);
+                        let pos = positions.get(i);
+                        if let Some(file) = file
+                            && let Some(pos) = pos
+                            && file == url
+                        {
+                            delete_rows.push(pos);
+                        }
+                    }
```

Comment:
Small nit: you can convert `file_paths` and `positions` into iterators and zip together, might be cleaner than this for loop since you don't use `i` for anything other than indexing.